### PR TITLE
Update Versions

### PR DESCRIPTION
**Pull request by michielderoos**
**Describe the Pull Request**

Update SDK target  

**Todo**

- [ ] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoMobileApp/issues) to the pull request
- [ ] Request review from relevant @person or team
- [ ] QA your pull request. This includes running the app, login, testing changes, [accessibility](https://reactnative.dev/docs/accessibility) etc.
- [ ] Bump [package](https://github.com/teamsempo/SempoMobileApp/blob/master/package.json#L3) and [android](https://github.com/teamsempo/SempoMobileApp/blame/master/android/app/build.gradle#L141-L142) versions following Semver
- [ ] Add release notes to [unreleased changelog](https://github.com/teamsempo/SempoMobileApp/blob/master/CHANGELOG.md#unreleased)

### CODE REVIEW

**Comment by michielderoos** [comment]
[Mobile app version bumps](https://trello.com/c/kQw6KSzL/936-mobile-app-version-bumps)

File: android/build.gradle [hunk 1]
```diff
@@ -4,7 +4,7 @@ buildscript {
     ext {
         buildToolsVersion = "29.0.5"
         googlePlayServicesVersion = "12.0.1"
-        minSdkVersion = 16
+        minSdkVersion = 21
         compileSdkVersion = 29
         targetSdkVersion = 29
```
Comment (tristanhcole) [inline android/build.gradle:9]: API level requirement `Android 11 (API level 30)`. November 1, 2021: Required for app updates
https://support.google.com/googleplay/android-developer/answer/9859152?visit_id=637785872990684562-2732748939&rd=1#targetsdk&zippy=%2Ctarget-api-level-requirements-for-play-console

**Review by tristanhcole — CHANGES_REQUESTED** [review]
When I try to build a release version of the app
```
cd android
./gradlew bundleRelease
```

The following error shows

```
> Task :react-native-nfc-manager:compileReleaseJavaWithJavac FAILED
/Users/tristancole/WebstormProjects/SempoMobileApp/node_modules/react-native-nfc-manager/android/src/main/java/community/revteltech/nfc/NfcManager.java:10: error: package android.support.annotation does not exist
import android.support.annotation.Nullable;
                                 ^
/Users/tristancole/WebstormProjects/SempoMobileApp/node_modules/react-native-nfc-manager/android/src/main/java/community/revteltech/nfc/NfcManager.java:1030: error: cannot find symbol
                           @Nullable WritableMap params) {
                            ^
  symbol:   class Nullable
  location: class NfcManager
2 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-nfc-manager:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.
```

File: android/build.gradle (hunk 1 above)
Comment (michielderoos) [reply android/build.gradle]: Bumped!

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -22,6 +22,7 @@
       <activity
```
Comment (tristanhcole) [inline android/app/src/main/AndroidManifest.xml:22]: https://developer.android.com/about/versions/12/behavior-changes-12#exported

File: package.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "clean": "rm -rf node_modules",
     "clean:android": "cd android &&./gradlew clean",
     "reinstall": "npm run clean && npm install",
+    "postinstall": "npx jetify",
```
Comment (tristanhcole) [inline package.json:13]: https://github.com/invertase/react-native-firebase/issues/2332

**Review by tristanhcole — APPROVED** [review]
Good to go! Tested and finally resolved outstanding issues

**Pull request merged**